backend/src/routes/helpers/validation.ts

```diff
@@ -0,0 +1,9 @@
+import * as z from 'zod';
+
+export const stringToNumberTransformer = z.transformer(
+  z.string().nonempty(),
+  z.number(),
+  (val) => {
+    return +val;
```

Comment:
I prefer using explicit `Number` conversion when converting between types.

Reply:
Will use Number in subsequent PRs